dht/partition_filter.hh

```diff
@@ -28,10 +29,10 @@ public:
         // While token T is after a range Rn, advance the iterator.
         // iterator will be stopped at a range which either overlaps with T (if T belongs to node),
         // or at a range which is after T (if T doesn't belong to this node).
-        while (_it != _sorted_owned_ranges.end() && _it->after(t, dht::token_comparator())) {
-            _it++;
-        }
-
+        _it = std::lower_bound(_it, _sorted_owned_ranges.end(), t,
+                               [] (const dht::token_range& range, const dht::token& token) {
+            return range.after(token, dht::token_comparator());
+        });
```

Comment:
This is a pessimization. t usually crawls slowly upwards so the original code terminates without looping. std::lower_bound will run O(log n) iterations and end up not changing _it.

Reply:
you mean, `t` in general just increments for only a single time, and very likely reaches the lower_bound right away?